_posts/2019-02-27-stripe-connect-part-1.md

```diff
@@ -0,0 +1,187 @@
+      query: {
+        client_secret: ENV["STRIPE_SECRET_KEY"],
+        code: params[:code],
+        grant_type: "authorization_code"
+      }
+    )
+  end
+end
+```
+### Add `stripe_user_id` to our database
+
+We should have the `stripe_user_id` column in our users table so we can connect the Stripe accounts to our User objects.
+
+```ruby
+# db/migrate/20190227164333_add_stripe_user_id_to_users.rb
+
+class AddStripeUserIdToUsers < ActiveRecord::Migration[5.2]
+  def change
+    add_column :users, :stripe_user_id, :string
+  end
+end
+```
+
+### Save the `stripe_user_id` and show a message to the user
+
+Finally if the response was successful we redirect with a message and we also update the User with the `stripe_user_id` that we received.
+
+```ruby
+# app/controllers/stripe_controller.rb
+
+class StripeController < ApplicationController
+  def connect
+    response = HTTParty.post("https://connect.stripe.com/oauth/token",
+      query: {
+        client_secret: ENV["STRIPE_SECRET_KEY"],
+        code: params[:code],
+        grant_type: "authorization_code"
+      }
+    )
+
+    if response.parsed_response.key?("error")
+      redirect_to user_settings_path(user),
+        notice: response.parsed_response["error_description"]
+    else
+      stripe_user_id = response.parsed_response["stripe_user_id"]
+      @user.update_attribute(:stripe_user_id, stripe_user_id)
+
+      redirect_to user_settings_path(user),
+        notice: 'User successfully connected with Stripe!'
+    end
+  end
+end
+```
+
+Now that we have a `stripe_user_id` we can show a message instead of the "Connect with Stripe" button:
+
+```ruby
+# app/views/users/settings.html.erb
+
+<% if @user.stripe_user_id %>
+  Connected with Stripe
+<% else %>
+  <%= link_to image_tag("stripe_connect_button.png"), stripe_connect_link %>
+<% end %>
+```
+
+### Bonus
+
+Stripe offers your connected accounts to have access to a [dashboard](https://stripe.com/docs/connect/express-dashboard) where they can see their balance and some of the information they entered during the setup. You can include a link to it so your users can access there.
+
+```ruby
+# app/views/users/settings.html.erb
+
+<% if @user.stripe_user_id %>
+  <%= link_to "Go to Stripe Dashboard", stripe_dashboard_path(@user.id) %>
+<% end %>
+```
+
+```ruby
+# routes.rb
+
+get "stripe/dashboard/:user_id", to: "stripe#dashboard", as: :stripe_dashboard
+```
+
+You'll need to install the [stripe gem](https://github.com/stripe/stripe-ruby) if you haven't installed it yet.
+
+```ruby
+class StripeController < ApplicationController
+  def dashboard
+    account = Stripe::Account.retrieve(@user.stripe_user_id)
+    login_links = account.login_links.create
+
+    redirect_to login_links.url
+  end
+end
+```
+
+### Conclusion
+
+This first part was just to be able to connect Stripe accounts to your users. Stay tuned for the next part where we'll be talking about the payments, charges, fees and more!
```

Comment:
You might want to number some of these steps (1., 2., 3., etc.) like in the Stripe article you link to. It will make it more clear.

Reply:
@ercohen14 hmm, maybe that will make it more confused because I'm not following the exact same order of steps that the Stripe documentation.